src/App.js

```diff
@@ -19,20 +31,31 @@ function PokeApp() {
         type: "",
     });
 
+
+    const updateUrl = (lowerCasePokemonName) => {
+        navigate(`/${lowerCasePokemonName}`, { replace: true });
+    };
+
     const searchPokemon = () => {
         const lowerCasePokemonName = pokemonName.toLowerCase();
         axios
             .get(`https://pokeapi.co/api/v2/pokemon/${lowerCasePokemonName}`)
             .then((response) => {
+                const weightInKg = Math.round(response.data.weight / 10)
+                const heightInMetres = Math.round((response.data.height / 10) * 10) / 10;
+                const titleCasePokemonName = toTitleCase(pokemonName)
                 setPokemonInfo({
-                    name: pokemonName,
+                    name: titleCasePokemonName,
                     species: response.data.species.name,
                     imageUrl: response.data.sprites.front_default,
                     hp: response.data.stats[0].base_stat,
                     attack: response.data.stats[1].base_stat,
                     defense: response.data.stats[2].base_stat,
                     type: response.data.types[0].type.name,
+                    weight: weightInKg,
+                    height: heightInMetres,
                 });
+                updateUrl(lowerCasePokemonName)
```

Comment:
Instead of manually updating the URL, why not just make the Pokemon details a Link? That way you don't have to write any code to update the URL you can just let the user click the Link and the platform will deal with updating the URL and your code can just respond to that happening. 

React Router [has a Link component](https://reactrouter.com/en/main/components/link#link) that'll work.


i.e instead of having the `div` in [this bit of the page](https://github.com/d-elv/pokeApi/pull/2/files#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67R113) you could have a `Link`, something like:

```
<Link to={pokemonInfo.name}>
    // ...rest of the details stuff
</Link>
```

Reply:
smashed it.